lib/src/migrators/module/references.dart

```diff
@@ -300,21 +311,27 @@ class _ReferenceVisitor extends RecursiveAstVisitor {
       if (import is DynamicImport) {
         var result =
             importCache.import(Uri.parse(import.url), _importer, _currentUrl);
-        if (result != null) {
-          var oldImporter = _importer;
-          _importer = result.item1;
-          visitStylesheet(result.item2);
-          var url = result.item2.span.sourceUrl;
-          var importSource = ImportSource(url, import);
-          for (var declaration in _declarationSources.keys.toList()) {
-            var source = _declarationSources[declaration];
-            if (source.url == url &&
-                (source is CurrentSource || source is ForwardSource)) {
-              _declarationSources[declaration] = importSource;
-            }
+        if (result == null) return;
+
+        // TODO(nweiz): Rather than skipping a stylesheet we've already visited,
+        // visit it again and throw an error if it produces different results
+        // (#99).
+        var url = result.item2.span.sourceUrl;
+        if (_importers.containsKey(url)) return;
```

Comment:
We need to revisit duplicate imports in `_ReferenceVisitor` (but not necessarily in `_ModuleMigrationVisitor`) to avoid breaking cases like this:

```
<==> entrypoint.scss
@import "a";
@import "b";

a {
  color: fn();
}

<==> _a.scss
@import "b";
@function fn() {@return blue;}

<==> _b.scss
@function fn() {@return green;}
```

Pre-migration, `fn()` within `entrypoint` should refer to the one in `b`, but if we only visit `b` the first time it's encountered, the definition in `a` will override it by the time the reference is visited and the main migrator will then use the wrong namespace.

Reply:
A good example of why it's important to have thorough test cases :smile:! Done.